amethyst_core/src/named.rs

```diff
@@ -1,6 +1,160 @@
-use std::borrow::Cow;
+use std::{borrow::Cow, ops::Deref};
 
-use specs::{world::LazyBuilder, Component, DenseVecStorage, EntityBuilder, WriteStorage};
+use fnv::FnvHashMap as HashMap;
+use specs::{
+    storage::MaskedStorage, world::LazyBuilder, Component, DenseVecStorage, Entity, EntityBuilder,
+    Storage, WriteStorage,
+};
+
+use util::{Cache, CachedStorage};
+
+/// Extension trait to find an entity based on its `Named` component.
+/// Can only be used on the `Named` component storage.
+pub trait FindNamed {
+    /// Finds an entity by its name by querying it from a map.
+    ///
+    /// For performance reasons, this should only be used once, and the returned `Entity` shall
+    /// be stored to avoid future lookups.
```

Comment:
Might reword to "this is a mildly expensive operation, so please avoid it calling for large groups of entities every frame.  To avert this, you can store the returned `Entity` value in a resource or component."